src/test/scala/cromwell/engine/db/slick/SlickDataAccessSpec.scala

```diff
@@ -213,6 +221,122 @@ class SlickDataAccessSpec extends FlatSpec with Matchers with ScalaFutures {
+        _ = params.workflowId shouldBe workflowInfo.id
+        _ = params.callKey shouldEqual Option(ExecutionDatabaseKey("three_step.ps", None))
+        _ <- dataAccess.updateCallCaching(params)
+        executions <- dataAccess.getExecutions(workflowInfo.id)
+        (allowing, disallowing) = executions partition { _.allowsResultReuse }
+        _ = allowing should have size 2
+        _ = disallowing should have size 1
+        _ = disallowing.seq.head.callFqn should be ("three_step.ps")
+
+        // The `inside_scatter` is a to-be-exploded placeholder, but it will conflict with the collector that the
+        // scatter explodes below so filter that out.
+        _ <- dataAccess.createWorkflow(scatterWorkflowInfo, Nil, scatterWorkflowInfo.namespace.workflow.calls.filterNot(_.name == "inside_scatter"), localBackend)
+        scatter = scatterWorkflowInfo.namespace.workflow.scatters.head
+        scatterKey = ScatterKey(scatter, None)
+        newEntries = scatterKey.populate(5)
+        _ <- dataAccess.insertCalls(scatterWorkflowInfo.id, newEntries.keys, localBackend)
+        executions <- dataAccess.getExecutions(scatterWorkflowInfo.id)
+        _ = executions foreach { _.allowsResultReuse shouldBe true }
+
+        outsideParams <- CallCachingParameters.from(scatterWorkflowInfo.id, Option("scatter0.outside_scatter"), AllowFalse, dataAccess)
+        _ <- dataAccess.updateCallCaching(outsideParams)
+        executions <- dataAccess.getExecutions(scatterWorkflowInfo.id)
+        (allowing, disallowing) = executions partition { _.allowsResultReuse }
+        _ = allowing should have size (executions.size - 1)
+        _ = disallowing should have size 1
+        _ = disallowing.seq.head.callFqn should be ("scatter0.outside_scatter")
+
+        // Support unindexed scattered call targets to update all shards
+        _ = executions filter { _.callFqn == "scatter0.inside_scatter" } foreach { _.allowsResultReuse shouldBe true }
+        unindexedCallParams <- CallCachingParameters.from(scatterWorkflowInfo.id, Option("scatter0.inside_scatter"), AllowFalse, dataAccess)
+        _ <- dataAccess.updateCallCaching(unindexedCallParams)
+        executions <- dataAccess.getExecutions(scatterWorkflowInfo.id)
+        _ = executions filter { _.callFqn == "scatter0.inside_scatter" } foreach { _.allowsResultReuse shouldBe false }
+
+        insideParams <- CallCachingParameters.from(scatterWorkflowInfo.id, Option("scatter0.inside_scatter.3"), AllowTrue, dataAccess)
+        _ <- dataAccess.updateCallCaching(insideParams)
+        executions <- dataAccess.getExecutions(scatterWorkflowInfo.id)
+        inside = executions filter { e => e.callFqn == "scatter0.inside_scatter" }
+        (allowing, disallowing) = inside partition { _.allowsResultReuse }
+        _ = allowing should have size 1
+        _ = disallowing should have size (inside.size - 1)
+      } yield ()).futureValue
+    }
```

Comment:
This might be my personal failing but these tests seem rather complex and difficult to follow. Can anything be removed into easier-to-follow chunks?

Also they both define the same private internal method so that's a good candidate for being extracted first.